cmd/queue/main.go

```diff
@@ -420,15 +427,25 @@ func buildAdminServer(healthState *health.State, logger *zap.SugaredLogger) *htt
 	}
 }
 
-func buildMetricsServer(promStatReporter *queue.PrometheusStatsReporter) *http.Server {
+func buildMetricsServer(promStatReporter *queue.PrometheusStatsReporter, protobufStatReporter *queue.ProtobufStatsReporter) *http.Server {
 	metricsMux := http.NewServeMux()
-	metricsMux.Handle("/metrics", promStatReporter.Handler())
+	metricsMux.Handle("/metrics", metricsHttpHandler(promStatReporter, protobufStatReporter))
 	return &http.Server{
 		Addr:    ":" + strconv.Itoa(networking.AutoscalingQueueMetricsPort),
 		Handler: metricsMux,
 	}
 }
 
+func metricsHttpHandler(promStatReporter *queue.PrometheusStatsReporter, protobufStatReporter *queue.ProtobufStatsReporter) http.Handler {
+	return http.HandlerFunc(func(rsp http.ResponseWriter, req *http.Request) {
+		if strings.Contains(req.Header.Get("Accept"), network.ProtoAcceptContent) {
```

Comment:
Potentially this could be a list of accepted contents, so it is more technically correct to do it this way. However, in this context it should be one type of content in general, because prometheus text format didnt require to set anything.